dotcom-rendering/src/lib/assets.ts

```diff
@@ -47,31 +37,63 @@ const isDev = process.env.NODE_ENV === 'development';
 
 export const ASSET_ORIGIN = decideAssetOrigin(process.env.GU_STAGE, isDev);
 
-export const getScriptArrayFromFile = (
+const getManifest = (path: string): AssetHash => {
+	try {
+		const assetHash: unknown = JSON.parse(
+			readFileSync(resolve(__dirname, path), { encoding: 'utf-8' }),
+		);
+		if (typeof assetHash != 'object' || assetHash === null)
+			throw new Error('Not a valid AssetHash type');
+
+		/** @TODO validate the object */
+		return assetHash as AssetHash;
+	} catch (e) {
+		console.error('Could not load manifest in: ', path);
+		console.error('Some filename lookups will fail');
```

Comment:
Would this produce console errors in `make dev` as the legacy bundle does not exist?

Reply:
No, because we’re not using the manifests in development mode.

We’re returning early on lines 66-71 inside `getScripts` if `isDev`.